Assets/AnkrSDKImporter/Editor/AnkrSDKImporter.cs

```diff
@@ -0,0 +1,205 @@
+      private const string OpenUpmRegistryUrl = "https://package.openupm.com";
+      
+      private static ListRequest _listRequest;
+   
+      static AnkrSDKImporter()
+      {
+         if (Application.companyName == CompanyNameToRejectImport
+             && Application.productName == ProductNameToRejectImport) 
+            return;
+         
+         _listRequest = Client.List();
+         EditorApplication.update += EditorApplicationOnUpdate;
+      }
+
+      private static void EditorApplicationOnUpdate()
+      {
+         if (_listRequest.IsCompleted)
+         {
+            if (_listRequest.Status == StatusCode.Success)
+            {
+               var filteredPackagesToImport = new List<PackageData>();
+               foreach (PackageData packageData in PackagesToTryToImport)
+               {
+                  if (PackageCollectionContains(_listRequest.Result, packageData))
+                     continue;
+                  
+                  filteredPackagesToImport.Add(packageData);
+               }
+               
+               AddDataToManifest(filteredPackagesToImport);
+            }
+            else
+            {
+               Debug.Log("AnkrSDKImporter: Could not check for packages: " + _listRequest.Error.message);
+            }
+
+            EditorApplication.update -= EditorApplicationOnUpdate;
+            _listRequest = null;
+         }
+      }
+
+      private static bool PackageCollectionContains(PackageCollection collection, PackageData packageData)
+      {
+         if (packageData.External)
+         {
+            //if package is external (meaning loaded by URL) we only consider it existing if the package id is already present
+            //in the current project package collection
+            foreach (PackageInfo package in collection)
+            {
+               if (package.packageId == packageData.PackageId)
+               {
+                  return true;
+               }
+            }
+         }
+         else
+         {
+            //if project is part of unity package registry we need to make sure that the version is the same
+            //otherwise we consider it not present in the project and manifest modification will
+            //just change the version in the manifest dependencies json object to make sure 
+            //versions for all dependencies will be the ones specified in this class
+            foreach (PackageInfo package in collection)
+            {
+               if (package.packageId == packageData.PackageId && package.version == packageData.PackageVersionOrUrl)
+               {
+                  return true;
+               }
+            }
+         }
+
+         return false;
+      }
+
+      private static void AddDataToManifest(List<PackageData> packagesToImport)
+      {
+         string assetsFolderPath = Application.dataPath;
+         string packagesFolderPath = assetsFolderPath.TrimEnd(new [] { '/'})
+            .TrimEnd(new [] { '\\'})
```

Comment:
can we use "File"/"FileInfo"+Path class instead? Mac and windows has different "//" style for example